src/mon/MonClient.h

```diff
@@ -483,12 +487,27 @@ class MonClient : public Dispatcher {
    * get latest known version(s) of cluster map
    *
    * @param map string name of map (e.g., 'osdmap')
-   * @param newest pointer where newest map version will be stored
-   * @param oldest pointer where oldest map version will be stored
-   * @param onfinish context that will be triggered on completion
+   * @param token context that will be triggered on completion
    * @return (via context) 0 on success, -EAGAIN if we need to resubmit our request
    */
-  void get_version(string map, version_t *newest, version_t *oldest, Context *onfinish);
+  template<typename CompletionToken,
+	   typename Signature = VersionSig,
+	   typename Handler = typename asio::handler_type<CompletionToken,
+							  Signature>::type,
+	   typename Result = typename asio::async_result<Handler>::type>
+  Result get_version(string map, CompletionToken&& token) {
```

Comment:
probably worth changing `map` to a const ref here. otherwise, use move-assignment for `m->what = map;` below